src/MSBuild/TerminalLogger/Terminal.cs

```diff
@@ -62,6 +63,12 @@ public int Width
         }
     }
 
+    /// <inheritdoc/>
+    /// <remarks>
+    /// https://github.com/dotnet/msbuild/issues/8958: iTerm2 treats ;9 code to post a notification instead, so disable progress reporting on Mac.
+    /// </remarks>
+    public bool SupportsProgressReporting { get; } = !RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
```

Comment:
```suggestion
    public bool SupportsProgressReporting { get; } = Environment.GetEnvironmentVariable("TERM_PROGRAM") != "iTerm.app";
```
If it's just iTerm2 perhaps this? (feels super hacky I know)

Reply:
Just tried a few, and none of them appear to support it:
* Terminal (built in)
* iTerm2
* Hyper
* Alacritty